README.md

```diff
@@ -330,6 +333,94 @@ RUN go build -o hello hello.go && \
     xx-verify hello
 ```
 
+## Rust
+
+Building Rust can be achieved with the `xx-cargo` wrapper that automatically
+sets up the target triple and also `pkg-config` and C compiler.
+
+The wrapper supports rust installed via [`rustup`](https://rustup.rs/)
+(alpine/debian), distribution packages (alpine/debian) and the [official `rust` image](https://hub.docker.com/_/rust).
+
+### Building on Alpine
+
+```dockerfile
+# official rust image
+FROM --platform=$BUILDPLATFORM rust:alpine
+RUN apk add clang lld
+# ...
+ARG TARGETPLATFORM
+RUN xx-apk add xx-c-essentials
+RUN xx-cargo build --release --target-dir ./build && \
+    xx-verify ./build/$(xx-cargo --print-target)/release/hello_cargo
+```
+
+```dockerfile
+# rustup
+FROM --platform=$BUILDPLATFORM alpine AS rustup
+RUN apk add curl
+RUN curl https://sh.rustup.rs -sSf | sh -s -- -y --default-toolchain stable --no-modify-path --profile minimal
+ENV PATH="/root/.cargo/bin:$PATH"
+
+FROM rustup
+RUN apk add clang lld
+# ...
```

Comment:
Maybe we should also provide (a link to) a full `b3sum` build. I can see it could be somewhat confusing about how to fetch cargo packages so it is not duplicated.

Reply:
This one looks good to you as an example: https://github.com/crazy-max/rust-docker-cross/blob/main/Dockerfile ?

Reply:
It could use some comments and I'm not sure if all of these cache mounts are safe.

Maybe we can just provide an example of the `cargo fetch` as a separate block in xx readme.